muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/guider/dialogs/student/tabs/contact-events/editors/new-event.tsx

```diff
@@ -0,0 +1,355 @@
+   */
+  constructor(props: NewContactEventProps) {
+    super(props, "new-contact-event");
+
+    this.state = this.getRecoverStoredState({
+      text: props.initialMessage || "",
+      sender: props.initialSender || "",
+      date:
+        props.i18n.time.getLocalizedMoment(props.initialDate).toDate() ||
+        props.i18n.time.getLocalizedMoment().toDate(),
+      type: props.initialType || "OTHER",
+      locked: false,
+    });
+  }
+
+  /**
+   * checkAgainstStoredState
+   */
+  checkAgainstStoredState = () => {
+    this.checkStoredAgainstThisState(
+      {
+        text: this.props.initialMessage || "",
+        sender: this.props.initialSender || "",
+        date:
+          this.props.i18n.time
+            .getLocalizedMoment(this.props.initialDate)
+            .toDate() || this.props.i18n.time.getLocalizedMoment().toDate(),
+        type: this.props.initialType || "OTHER",
+        locked: false,
+      },
+      this.nameSpace
+    );
+
+    this.props.onOpen && this.props.onOpen();
+  };
+
+  /**
+   * onCKEditorChange handler
+   * @param text ckeditor text
+   */
+  onCKEditorChange = (text: string) => {
+    if (this.avoidCKEditorTriggeringChangeForNoReasonAtAll) {
+      this.avoidCKEditorTriggeringChangeForNoReasonAtAll = false;
+      return;
+    }
+    this.setStateAndStore({ text }, this.nameSpace);
+  };
+
+  /**
+   * onDateChange handler
+   * @param date Date
+   */
+  onDateChange = (date: Date) => {
+    this.setStateAndStore({ date: date }, this.nameSpace);
+  };
+
+  /**
+   * onTypeChange handler
+   * @param e event
+   */
+  onTypeChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    this.setStateAndStore(
+      { type: e.target.value as ContactTypes },
+      this.nameSpace
+    );
+  };
+
+  /**
+   * saveContactEvent
+   * @param closeDialog closeDialog
+   */
+  saveContactEvent = (closeDialog: () => void) => {
+    this.setState({
+      locked: true,
+    });
+    this.props.createContactLogEvent(
+      this.props.currentStudent.userEntityId,
+      /**
+       * payload
+       */
+      {
+        text: this.state.text,
+        entryDate: moment(this.state.date).format(),
+        type: this.state.type,
+      },
+      /**
+       * onSuccess
+       */
+      () => {
+        this.setState({
+          locked: false,
+        });
+        closeDialog();
+        this.clearUp();
+      }
+    );
+  };
+
+  /**
+   * clearUp
+   */
+  clearUp = () => {
+    this.avoidCKEditorTriggeringChangeForNoReasonAtAll = true;
+    setTimeout(() => {
+      this.avoidCKEditorTriggeringChangeForNoReasonAtAll = false;
+    }, 100);
+    this.setStateAndClear({
+      text: this.props.initialMessage || "",
+      sender: this.props.initialSender || "",
+      date:
+        this.props.i18n.time
+          .getLocalizedMoment(this.props.initialDate)
+          .toDate() || this.props.i18n.time.getLocalizedMoment().toDate(),
+      type: this.props.initialType || "OTHER",
+      locked: false,
```

Comment:
Is missing `this.namespace` as it holds the id value which is used to clear localStorage